tagstudio/src/qt/widgets/preview_panel.py

```diff
@@ -130,21 +139,48 @@ def __init__(self, library: Library, driver: "QtDriver"):
         self.open_file_action = QAction("Open file", self)
         self.open_explorer_action = QAction(PlatformStrings.open_file_str, self)
 
+        self.base_preview_type = previewType.IMG
+
         self.preview_img = QPushButtonWrapper()
         self.preview_img.setMinimumSize(*self.img_button_size)
         self.preview_img.setFlat(True)
         self.preview_img.setContextMenuPolicy(Qt.ContextMenuPolicy.ActionsContextMenu)
         self.preview_img.addAction(self.open_file_action)
         self.preview_img.addAction(self.open_explorer_action)
 
-        self.preview_gif = QLabel()
-        self.preview_gif.setMinimumSize(*self.img_button_size)
-        self.preview_gif.setContextMenuPolicy(Qt.ContextMenuPolicy.ActionsContextMenu)
-        self.preview_gif.setCursor(Qt.CursorShape.ArrowCursor)
-        self.preview_gif.addAction(self.open_file_action)
-        self.preview_gif.addAction(self.open_explorer_action)
-        self.preview_gif.hide()
-        self.gif_buffer: QBuffer = QBuffer()
+        self.preview_anim_img = QLabel()
+        self.preview_anim_img.setMinimumSize(*self.img_button_size)
+        self.preview_anim_img.setContextMenuPolicy(Qt.ContextMenuPolicy.ActionsContextMenu)
+        self.preview_anim_img.setCursor(Qt.CursorShape.ArrowCursor)
+        self.preview_anim_img.addAction(self.open_file_action)
+        self.preview_anim_img.addAction(self.open_explorer_action)
+        self.preview_anim_img.hide()
+        self.anim_img_buffer: QBuffer = QBuffer()
+
+        self.preview_anim_img_fmts = []
+
+        qmovie_formats = QMovie.supportedFormats()
+
+        self.preview_anim_img_fmts = [fmt.data().decode("utf-8") for fmt in qmovie_formats]
+
+        ani_img_priority_order = ["jxl", "apng", "png", "webp", "gif"]
+
+        self.preview_anim_img_pil_map = {"apng": "png"}
+
+        self.preview_anim_img_pil_map_args = {"gif": {"disposal": 2}}
+
+        self.preview_anim_img_pil_known_good = {"webp", "gif"}
```

Comment:
There was a lot of this code that I opted to leave out when porting the functionality over due to it being difficult to decipher the purpose of and to trace back what was occurring inside the animated image creation block.

If it's necessary for important additional functionality, I would appreciate some documentation on what exactly all of these maps are being used for and why they need to be used over just loading in the image as a gif/webp.

Reply:
@BPplays this is for choosing image format priority
- ani_img_priority_order: is an order of image extensions that are sorted first if they're supported by both pillow and qmovie
- preview_anim_img_pil_map: this maps what extension to use in pillow, pillow encodes animated pngs with the format 'png' instead of 'apng' so if apng comes up anywhere pillow should use png
- preview_anim_img_pil_map_args: this is special args for pillow should use for specific encoding formats, here gif has a trail on transparency if this is not used, but webp does not need it.
- preview_anim_img_pil_known_good: it's a list of known image types pillow can encode with animation, i couldn't find a way to test this in the program, ~it also probably needs expanding~ i made a commit